components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -1386,3 +1386,70 @@ func (backend *ESClient) getDocFromNodeRunInfoFromNodeId(ctx context.Context, no
 	}
 	firstRun <- item.FirstRun
 }
+
+//GetReportsDayLatestTrue Get the Nodes from past 90 days from current date for upgrading
+func (backend *ESClient) GetReportsDayLatestTrue(ctx context.Context, time90daysAgo time.Time) ([]relaxting.ESInSpecReport, error) {
+	reportList := make([]relaxting.ESInSpecReport, 0)
+
+	indicesSlice, err := get90DaysIndex(time90daysAgo)
+	if err != nil {
+		logrus.Errorf("Unable to get the indexe for last 90 days %v", err)
+		return nil, err
+	}
+
+	boolQuery := elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery("day_latest", true))
+
+	fsc := elastic.NewFetchSourceContext(true)
+
+	searchSource := elastic.NewSearchSource().
+		FetchSourceContext(fsc).
+		Query(boolQuery).
+		Size(10000).Sort("end_time", false)
+
+	for _, index := range indicesSlice {
+		searchResult, err := backend.client.Search().
+			SearchSource(searchSource).
+			Index(index).
+			FilterPath(
+				"took",
+				"hits.total",
+				"hits.hits._id",
+				"hits.hits._source").
+			Do(ctx)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if searchResult.TotalHits() > 0 {
+			// Iterate through results
+			for _, hit := range searchResult.Hits.Hits {
+				var report relaxting.ESInSpecReport
+				if hit.Source != nil {
+					err := json.Unmarshal(hit.Source, &report)
+					if err != nil {
+						logrus.Errorf("Received error while unmarshling for reports in upgrade for comp run info %+v", err)
+					}
+
+				}
+				reportList = append(reportList, report)
```

Comment:
Are we storing the complete report structure in this list? That is going to be huge